### ISSUE TITLE
Registration function

### ISSUE THREAD

**Issue by qy21gafy** [amosproj/amos2022ws05-shared-desk-mgmt#26]
## User story
1. As a new user
2. I want / need to be able to register for the system with my data
3. So that I can login later and access the features of shared desk management

## Acceptance criteria
- [x] Gets user input: email, name, password, company id
- [x] Every user should only be able to register once with the same mail
- [x] After registration a user should be marked as a user that awaits approval
- [x] The registration should fail if the company is not yet registered
- [x] The provided data should be stored in the database if the registration doesn't fail
- [x] Registration function should return if the registration was successful or not

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
- [ ] Feature has been fully implemented
- [ ] Feature has been merged into the mainline
- [ ] All acceptance criteria were met
- [ ] Product owner approved features
- [ ] All tests are passing
- [ ] Developers agreed to release

**Comment by qy21gafy** [comment]
Later on add a issue that prevents simple attacks such as SQL injection

**Comment by stefanpfahler** [comment]
Does it really make sense for an admin to be able to create users without a company ID? Aren't the admins just meant to be admins of a company anyway?.. Maybe you can elaborate a little on this @qy21gafy

**Comment by stefanpfahler** [comment]
And also who exactly is the initial admin?

**Comment by qy21gafy** [comment]
We'll have to elaborate about how exactly we want to handle the registration on Wednesday. Our (the POs) current idea was to add an initial admin that is the person who first registers their company with the service.  This way the company is first added to the database. This is also explained in the product glossary. This user is granted admin priviledges from the beginning and has to be approved in some other way than normal users because no admin for that new company exists yet.  Normal users have to be approved by the admins of their company before they can access the system after registration. Normal users can be granted admin priviledges later on (for example if they are admins in their company as well).

**Comment by stefanpfahler** [comment]
Okay thank you for your fast reply! 
But I don't think we have time to discuss that topic in our weekly meeting..
We'll go ahead and implement the admin logic as you described in your comment above in the meantime if you think that's okay

**Comment by stefanpfahler** [comment]
If not we should alter the acceptance criteria so we don't run into complications in our weekly imo

**Comment by qy21gafy** [comment]
I think you're right and we should alter the acceptance criteria for now and figure out the initial registration in the next sprint, maybe by adding a new issue. I think the first bullet point can be updated by removing the initial admin input, the 4th bullet point can changed to: The registration should fail if the company is not yet registered with the service, the 5th bullet point can be removed. I think this is fine for the general registration process.

**Comment by stefanpfahler** [comment]
Sounds good! I will update this User Story as soon as we continue working on it 👍🏼
